builds/misc/metrics-collector-images-release.yaml

```diff
@@ -12,22 +12,20 @@ jobs:
     displayName: Metrics Collector Linux Build
     timeoutInMinutes: 30
     pool:
-      vmImage: 'ubuntu-18.04'
+      name: $(pool.linux.name)
+      demands:
+        - ImageOverride -equals agent-aziotedge-ubuntu-18.04-msmoby
```

Comment:
For the consistency, we should use `agent-aziotedge-ubuntu-18.04-docker` since that's what we used to build other docker images. Thank you @damonbarry for context.